lisa/runners/lisa_runner.py

```diff
@@ -290,30 +290,46 @@ def _prepare_environments(self) -> None:
     def _deploy_environment_task(
         self, environment: Environment, test_results: List[TestResult]
     ) -> None:
-        try:
+        max_retries = test_results[0].runtime_data.retry_on_deployment + 1
```

Comment:
The flag should be on environment, not test case. Every time, the retry count may applied to different test cases. So it cannot be set correctly.